### PR TITLE
Fix cypress view source lookup for invocations from within then()

### PR DESCRIPTION
**Pull request by ryanjduffy**
## Issue

When a cypress command is enqueued from with a `then()` callback, the call stack is different and we were not finding the correct frame to point to the invocation of the cypress command

## Resolution

Adjust the logic to not return a frame from within the cypress_runner

### CODE REVIEW

File: src/ui/components/TestSuite/suspense/TestStepSourceLocationCache.ts [hunk 1]
```diff
@@ -54,10 +54,34 @@ export const TestStepSourceLocationCache = createCacheWithTelemetry<
...
+                //     wrapper code like for cy.then in cypress_runner
+                //     __stackReplacementMarker in cypress_runner
+                //   ]
+                //
+                // We find it by looking from the top (the frames array comes in
+                // with the most recent last but it is reversed above) for the
+                // first entry that isn't from the cypress_runner where the
+                // _next_ frame is from the cypress_runner.
+
                 return l[i + 1]?.functionLocation?.some(fl => fl.sourceId === markerSourceId);
```
Comment (bvaughn) [inline src/ui/components/TestSuite/suspense/TestStepSourceLocationCache.ts:82]: I know you didn't newly add these variables in this PR, but I'd prefer we avoid variable names like `l` and `fl`. They're really confusing when you get nested deeply enough to not be able to see where they were defined.

**Review by bvaughn — COMMENTED** [review]
I'm really concerned about us encoding this level of internal Cypress details (potentially Cypress version specific details at that) in the frontend.

Can we create a Linear issue/project to move this out of the frontend?

**Comment by ryanjduffy** [comment]
> I'm really concerned about us encoding this level of internal Cypress details (potentially Cypress version specific details at that) in the frontend.
> 
> Can we create a Linear issue/project to move this out of the frontend?

We have FE-1354 and i'll add some comments to this effect in it.

File: src/ui/components/TestSuite/suspense/TestStepSourceLocationCache.ts (hunk 1 above)
Comment (ryanjduffy) [reply src/ui/components/TestSuite/suspense/TestStepSourceLocationCache.ts:82]: Sure. Can do!

**Review by bvaughn — APPROVED** [review]

**Pull request merged**